src/NovelRT/Input/NovelKey.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#include <NovelRT/Input/Input.h>
+#include <NovelRT/PluginManagement/PluginManagement.h>
+
+namespace NovelRT::Input
+{
+    NovelKey::NovelKey(std::string keyName, int32_t pairedKeyCode, int32_t modifier) noexcept
+    {
+        _keyName = keyName;
+        _pairedKey = pairedKeyCode;
+        _modifier = modifier;
+    }
+
+    void NovelKey::PairKey(int32_t externalKeyCode) noexcept
+    {
+        _pairedKey = externalKeyCode;
+    }
+
+    void NovelKey::UnpairKey() noexcept
+    {
+        _pairedKey = -1;
+    }
+
+    std::string& NovelKey::GetKeyName() noexcept
+    {
+        return _keyName;
+    }
+
+    int32_t NovelKey::GetExternalKeyCode() noexcept
+    {
+        return _pairedKey;
+    }
+
+    int32_t NovelKey::GetExternalModifierCode() noexcept
+    {
+        return _modifier;
+    }
```

Comment:
can we not make these const? Also why signed int?

Reply:
Can make const, however I based on int32_t based on both GLFW's and SDL's ways of typing key codes (both use a version of signed int32.

Is there a preference here? We can switch to unsigned and cast GLFW's keycodes internally for the glfw-based input device

Reply:
I just wonder what the benefit is, I guess. If its problematic then leave it. We need to get this in and I don't care enough. :^)

Reply:
That said - add const.